lib/Cake/Network/Email/CakeEmail.php

```diff
@@ -1717,6 +1721,27 @@ protected function _renderTemplates($content) {
 			$rendered[$type] = implode("\n", $rendered[$type]);
 			$rendered[$type] = rtrim($rendered[$type], "\n");
 		}
+
+		/* Embed images inline in html templates */
+		if (!empty($rendered['html'])) {
```

Comment:
I'm still not comfortable with this. The `print_r` could change the memory performance of an application. I'm also not sure this kind of automatic behavior belongs in a lower-level abstraction `Cake\Mailer\Email`. I could see this kind of behavior being made available through a plugin or a helper though.